ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcErrorResponse.java

```diff
@@ -21,15 +21,22 @@
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import com.google.common.base.MoreObjects;
 
-@JsonPropertyOrder({"jsonrpc", "id", "error"})
+@JsonPropertyOrder({"jsonrpc", "id", "error", "data"})
 public class JsonRpcErrorResponse implements JsonRpcResponse {
```

Comment:
To avoid adding all `data: null` in the test resorce file we can add:
`@JsonInclude(value = JsonInclude.Include.NON_NULL)`
So response will be build without any null field and I think will make more sense than returning a data=null. I wanted first to do it like this but I thought maybe you will want to have it returned.